### PR TITLE
2558 new module cnvkitgenemetrics

### LINKED ISSUE #2558: new module: cnvkit/genemetrics
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by marrip**
## PR checklist

Add `genemetrics` functionality to `cnvkit`.

Closes #2558

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/cnvkit/genemetrics/main.nf [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/cnvkit:0.9.9--pyhdfd78af_0':
+        'quay.io/biocontainers/cnvkit:0.9.9--pyhdfd78af_0' }"
+
+    input:
+    tuple val(meta), path(cnr), path(cns)
+
+    output:
+    tuple val(meta), path("*.tsv"), emit: tsv
+    //tuple val(meta), path("*.cnn"), emit: cnn
```
Comment (maxulysse) [inline modules/nf-core/cnvkit/genemetrics/main.nf:16]: why commented?

File: modules/nf-core/cnvkit/genemetrics/meta.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        Groovy Map containing sample information
+        e.g. [ id:'test', single_end:false ]
+  - cnr:
+      type: file
+      description: CNR file
+      pattern: "*.cnr"
+  - cns:
+      type: file
+      description: CNS file
+      pattern: "*.cns"
```
Comment (maxulysse) [inline modules/nf-core/cnvkit/genemetrics/meta.yml:26]: Given the code you wrote I'd say that this file is optional.
Can you confirm, and if so add this information here?

**Review by maxulysse — APPROVED** [review]
LGTM

File: modules/nf-core/cnvkit/genemetrics/main.nf (hunk 1 above)
Comment (marrip) [reply modules/nf-core/cnvkit/genemetrics/main.nf:16]: Sorry, my bad, it is not an output file from this process. I removed it ☺️

File: modules/nf-core/cnvkit/genemetrics/meta.yml (hunk 2 above)
Comment (marrip) [reply modules/nf-core/cnvkit/genemetrics/meta.yml:26]: you are completely right. Will add that information 😉

**Pull request merged**